docker/docker_tools.py

```diff
@@ -317,31 +358,52 @@ def build(
                     print(
                         f"Dev mode: since {bks} doesn't exist, cloning the BookServer..."
                     )
-                    xqt("git clone https://github.com/bnmnetp/BookServer.git")
+                    try:
+                        # Check if possible to clone BookServer with Custom Repo
+                        xqt("export GIT_TERMINAL_PROMPT=0 && git clone https://github.com/%s/BookServer.git"%clone_bs)
+                    except subprocess.CalledProcessError as e:
+                        # Clone with Default Repo if clone failure Detected (Git Error 128)
+                        xqt("git clone https://github.com/bnmnetp/BookServer.git")
                 rsc = Path("RunestoneComponents")
                 if not rsc.exists():
                     print(
                         f"Dev mode: since {rsc} doesn't exist, cloning the Runestone Components..."
                     )
-                    xqt(
-                        "git clone --branch peer_support https://github.com/RunestoneInteractive/RunestoneComponents.git"
-                    )
+                    try:
+                        # Check if possible to clone RunestoneComponents (peer_support) with Custom Repo
+                        xqt(
+                            "export GIT_TERMINAL_PROMPT=0 && git clone --branch peer_support https://github.com/%s/RunestoneComponents.git"%clone_bs
+                        )
+                    except subprocess.CalledProcessError as e:
+                        # Clone with Default Repo if clone failure Detected (Git Error 128)
+                        xqt(
+                            "git clone --branch peer_support https://github.com/RunestoneInteractive/RunestoneComponents.git"
+                        )
 
         # Ensure the user is in the ``www-data`` group.
         print("Checking to see if the current user is in the www-data group...")
         if "www-data" not in xqt("groups", capture_output=True, text=True).stdout:
             xqt('sudo usermod -a -G www-data "$USER"')
             did_group_add = True
             docker_sudo = True
+            
 
         # Provide this script as a more convenient CLI.
         xqt(
             f"{sys.executable} -m pip install --user -e docker",
         )
 
-        # Run the Docker build.
+        # Filter out clone flags
```

Comment:
Why do this? I'd suggest keeping all the flags, just in case we want to refer to them later.

Reply:
The only reason I removed the clone flags from being passed though is due to issues with the actual building of the docker containers. I might be able to replicate the issue.

Reply:
OK, let me know.

Reply:
After some testing, I never was able to replicate the error. I currently tested it with multiple combinations of the ```--clone-XX``` flag.